Common/Tests/Insights/HyperwalletInsightsTests.swift

```diff
@@ -30,43 +30,73 @@ class HyperwalletInsightsTests: XCTestCase {
+        let expectation = self.expectation(description: InsightsMock.trackClickExpectation)
+        insightsMock.expectations = [InsightsMock.trackClickExpectation: expectation]
+
         HyperwalletInsights.shared.trackClick(pageName: pageName, pageGroup: pageGroup, link: link, params: params)
-        XCTAssertTrue(mockInsights.didTrackClick, "HyperwalletInsights.trackClick should be called")
-        XCTAssertEqual(mockInsights.pageName, pageName, "Page name should be as expected")
-        XCTAssertEqual(mockInsights.pageGroup, pageGroup, "Page group should as expected")
-        XCTAssertEqual(mockInsights.link, link, "Link should be as expected")
-        XCTAssertNotNil(mockInsights.params[countryTag], "Params should have country")
-        XCTAssertNotNil(mockInsights.params[currencyTag], "Params should have currency")
+        wait(for: [expectation], timeout: 1)
+
+        XCTAssertTrue(insightsMock.didTrackClick, "HyperwalletInsights.trackClick should be called")
+        XCTAssertEqual(insightsMock.pageName, pageName, "Page name should be as expected")
+        XCTAssertEqual(insightsMock.pageGroup, pageGroup, "Page group should as expected")
+        XCTAssertEqual(insightsMock.link, link, "Link should be as expected")
+        XCTAssertNotNil(insightsMock.params[countryTag], "Params should have country")
+        XCTAssertNotNil(insightsMock.params[currencyTag], "Params should have currency")
+    }
+
+    func testTrackError() {
+        XCTAssertNotNil(HyperwalletInsights.shared, "HyperwalletInsights should be initialized")
+        let errorInfo = ErrorInfoBuilder(type: "errorInfo_type", message: "errorInfo_message")
+            .fieldName("errorInfo_fieldName")
+            .description("errorInfo_description")
+            .code("errorInfo_code")
+            .build()
+        let expectation = self.expectation(description: InsightsMock.trackErrorExpectation)
+        insightsMock.expectations = [InsightsMock.trackErrorExpectation: expectation]
+
+        HyperwalletInsights.shared.trackError(pageName: pageName, pageGroup: pageGroup, errorInfo: errorInfo)
+        wait(for: [expectation], timeout: 1)
+
+        XCTAssertTrue(insightsMock.didTrackError, "HyperwalletInsights.trackError should be called")
+        XCTAssertEqual(insightsMock.pageName, pageName, "Page name should be as expected")
+        XCTAssertEqual(insightsMock.pageGroup, pageGroup, "Page group should be as expected")
+        XCTAssertNotNil(insightsMock.errorInfo, "ErrorInfo shouldn't be empty")
     }
 
     func testTrackClick_InsightsNotInitialized() {
         hyperwalletInsights?.insights = nil
         XCTAssertNotNil(HyperwalletInsights.shared, "HyperwalletInsights should be initialized")
         let params = [countryTag: country, currencyTag: currency]
         HyperwalletInsights.shared.trackClick(pageName: pageName, pageGroup: pageGroup, link: link, params: params)
+        sleep(2)
```

Comment:
Why sleep instead of expectation?

Reply:
After the method `trackClick execution`, property `insights` is a real instance, not a mock.